tools/check_documentation.py

```diff
@@ -0,0 +1,170 @@
+
+    @classmethod
+    def expand_subclasses(cls):
+        """Recursively find subclasses"""
+        base_list = list(cls.BASE_SEARCH_CLASS.__subclasses__())
+        while len(base_list) > 0:
+            item = base_list[0]
+            base_list = base_list[1:]
+            cls.SUBCLASSES.append(item)
+            base_list.extend(item.__subclasses__())
+
+    @classmethod
+    def expand_exceptions(cls):
+        """Recursively find exceptions"""
+        base_list = cls.EXCEPTIONS
+        cls.EXCEPTIONS = []
+        while len(base_list) > 0:
+            item = base_list[0]
+            base_list = base_list[1:]
+            cls.EXCEPTIONS.append(item)
+            base_list.extend(item.__subclasses__())
+
+    @classmethod
+    def check(cls):
+        cls.expand_subclasses()
+        cls.expand_exceptions()
+        success = True  # Initially, there is success
+        for subclass in set(
+            list(
+                filter(cls.in_list_lambda(cls.EXCEPTIONS), cls.SUBCLASSES)
+            )  # Filters out all exceptions from the subclasses
+            + cls.EXPLICIT_INCLUDE
+            # adds the classes that should always be in the list
+        ):  # set conversion to exclude all duplicates
+            doc = subclass.__doc__  # extracts the documentation
+            if doc is None:  # Proceeds if there is no documentation
+                print(
+                    "Subclass {mod}.{name} is missing documentation.".format(
+                        mod=subclass.__module__, name=subclass.__name__
+                    )
+                )
+                success = False
+            else:  # Proceeds if there is documentation
+                has_attribute_table = bool(cls.HAS_ATTRIBUTE_TABLE.search(doc))
+                # does a regex search for attribute tables. Calling bool will
+                # make it False if there is no match and True if there is a
+                # match.
+                if not has_attribute_table:
+                    print(
+                        "Subclass {mod}.{name} is missing a table "
+                        "of common attributes.".format(
+                            mod=subclass.__module__, name=subclass.__name__
+                        )
+                    )
+                success &= has_attribute_table
+                # Sets success to False if there was not an attribute table,
+                # else keeps it True.
+            has_code_block = True
+            method_list = [
+                getattr(
+                    subclass, method_name
+                )  # Gets the method using the name
+                for method_name in filter(
+                    cls.in_list_lambda(cls.METHOD_EXCEPTIONS),
+                    dir(subclass),  # Gets list of attributes
+                )  # Filters out method exceptions
+                if (
+                    callable(
+                        getattr(subclass, method_name)
```

Comment:
Given that you need to call `getattr` up to 3 times in this list comprehension, I think this code would be better suited for a regular for loop.